orchestration/dagster_orchestration/hca_manage/tests/test_hca_manage.py

```diff
@@ -0,0 +1,190 @@
+            expected_target_blob.upload_from_file.assert_called_once_with(very_real_csv)
+
+    def test_process_rows_returns_zero_if_no_bad_rids(self):
+        result = self.manager._process_rows(
+            lambda: {'table_a', 'table_b'},
+            lambda table: set(),
+            soft_delete=True,
+            issue="oh dear")
+        self.assertEqual(result, 0)
+
+    def test_process_rows_does_no_soft_delete_if_soft_delete_false(self):
+        with patch('hca_manage.manage.HcaManage.soft_delete_rows') as mock_soft_delete:
+            result = self.manager._process_rows(
+                lambda: {'table_a', 'table_b'},
+                lambda table: {'abc'},
+                soft_delete=False,
+                issue="oh dear")
+            mock_soft_delete.assert_not_called()
+        self.assertEqual(result, 2)  # one error per table
+
+    def test_process_rows_does_soft_delete_if_soft_delete_true(self):
+        with patch('hca_manage.manage.HcaManage.soft_delete_rows') as mock_soft_delete:
+            result = self.manager._process_rows(
+                lambda: {'table_a', 'table_b'},
+                lambda table: {'abc'},
+                soft_delete=True,
+                issue="oh dear")
+            mock_soft_delete.assert_has_calls(
+                [
+                    call(StringContaining('table_a.csv'), 'table_a'),
+                    call(StringContaining('table_b.csv'), 'table_b'),
+                ],
+                any_order=True  # iteration order is not guaranteed for sets
+            )
+        self.assertEqual(result, 2)  # one error per table
+
+    def test_soft_delete_rows_calls_submit_soft_delete(self):
+        fake_google_path = 'fake://google/path.csv'
+        fake_job_id = 'jorb_id'
+        patched_soft_delete = patch('hca_manage.manage.HcaManage._submit_soft_delete', return_value=fake_job_id)
+        patched_put_in_bucket = patch('hca_manage.manage.HcaManage.put_csv_in_bucket', return_value=fake_google_path)
+        with patched_soft_delete as mock_submit, patched_put_in_bucket as mock_bucket_upload:
+            with NamedTemporaryFile() as fake_file:
+                returned_job_id = self.manager.soft_delete_rows(fake_file.name, 'table_a')
+            mock_bucket_upload.assert_called_once_with(
+                local_file=ObjectOfType(BufferedIOBase),
+                target_table='table_a',
+            )
+            mock_submit.assert_called_once_with(target_table='table_a', target_path=fake_google_path)
+            self.assertEqual(returned_job_id, fake_job_id)
+
+    def test_get_dataset_id_gets_id_of_first_dataset(self):
+        enumerate_datasets_response = Mock()
+        first_dataset = Mock()
+        first_dataset.id = 'abc'
+
+        second_dataset = Mock()
+        second_dataset.id = 'def'
+        enumerate_datasets_response.items = [first_dataset, second_dataset]
+        self.manager.data_repo_client.enumerate_datasets.return_value = enumerate_datasets_response
```

Comment:
Might be out of scope: the manager should raise if there is more than one dataset with the given name.